### PR TITLE
Make sure that hadoop is up and running before running product tests

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

**Comment by kokosing** [comment]
In our infrastructure (jenkins) we randomly got the tests failure that Presto is unable to connect the metastore. Let see if it solves our problem.

FYI: @ilfrin

File: presto-product-tests/bin/run_on_docker.sh [hunk 1]
```diff
@@ -240,6 +240,9 @@ PRESTO_LOGS_PID=$!
 # wait until presto is started
 retry check_presto
 
+# make sure that hadoop is still alive
+check_hadoop
```
Comment (sopel39) [inline presto-product-tests/bin/run_on_docker.sh]: `retry check_hadoop`?

File: presto-product-tests/bin/run_on_docker.sh [hunk 2]
```diff
@@ -46,7 +48,7 @@ function check_presto() {
   run_in_application_runner_container \
     java -jar "/docker/volumes/presto-cli/presto-cli-executable.jar" \
     ${CLI_ARGUMENTS} \
-    --execute "SHOW CATALOGS" | grep -i hive
+    --execute "SHOW CATALOGS" | grep -iq hive
```
Comment (ilfrin) [inline presto-product-tests/bin/run_on_docker.sh]: unrelated change?

File: presto-product-tests/bin/run_on_docker.sh [hunk 3]
```diff
@@ -238,6 +240,9 @@ PRESTO_LOGS_PID=$!
 # wait until presto is started
 retry check_presto
 
+# make sure that hadoop is still alive
+check_hadoop
```
Comment (ilfrin) [inline presto-product-tests/bin/run_on_docker.sh]: without retry this time?

File: presto-product-tests/bin/run_on_docker.sh (hunk 2 above)
Comment (kokosing) [reply presto-product-tests/bin/run_on_docker.sh]: Thanks to that change we `Avoid meaningless output from grep command`. But from the another grep command.

File: presto-product-tests/bin/run_on_docker.sh (hunk 3 above)
Comment (kokosing) [reply presto-product-tests/bin/run_on_docker.sh]: yes. We had retry previously, so hadoop should be up and running. No need to wait.

**Comment by kokosing** [comment]
> Why perform extra check for hadoop being alive.
I understand it could crash. But it could also crash during tests execution. It just does not look like the place where we perform extra check is special in anyway.

@losipiuk We have such health check implemented in benchto, maybe it would not be a bad idea to have similar thing in tempto. That way we would stop to run tests if any service (process) that tests depend on crashes. So far we do not have such thing. I added it as a best-effort, this is what I could today. However I am not sure if it helps. Do you think it would be better without it?

**Comment by ilfrin** [comment]
I confirmed on our internal testing that in fact this change does solve the problem that we've encountered: tests failing because can't reach the Hive Metastore, port 10000 closed.

**Comment by losipiuk** [comment]
> I've just confirmed on our internal testing that in fact this change does solve the problem that we've encountered: tests setup (requirements fulfilment) and tests failing because can't reach the Hive Metastore, port 10000 closed. It didn't crash, it just didn't come up in time.

Yeah - but I doubt it would not work without the last commit. That is the one which seems weird too me.
The first two are fine.

**Comment by kokosing** [comment]
@losipiuk Addressed your comment.
